pkgs/stdenv/darwin/default.nix

```diff
@@ -37,6 +37,7 @@ in rec {
     export CMAKE_OSX_ARCHITECTURES=x86_64
     # Workaround for https://openradar.appspot.com/22671534 on 10.11.
     export gl_cv_func_getcwd_abort_bug=no
+    export LC_CTYPE=UTF-8
```

Comment:
It would be still great if we could set LANG on macOS as well. Reason is, that `$LC_CTYPE` has a different semantics/priority compared to `$LANG`.  LANG has a lower priority then all `LC_*` variables, so if an existing package already sets `LC_ALL` or similar to some value, it would be preferred.

Reply:
It doesn't seem to work correctly:

```
$ LANG=UTF-8 locale
LANG="UTF-8"
LC_COLLATE="C"
LC_CTYPE="C"
LC_MESSAGES="C"
LC_MONETARY="C"
LC_NUMERIC="C"
LC_TIME="C"
LC_ALL=
```

vs.

```
$ LANG=en_US.UTF-8 locale
LANG="en_US.UTF-8"
LC_COLLATE="en_US.UTF-8"
LC_CTYPE="en_US.UTF-8"
LC_MESSAGES="en_US.UTF-8"
LC_MONETARY="en_US.UTF-8"
LC_NUMERIC="en_US.UTF-8"
LC_TIME="en_US.UTF-8"
LC_ALL=
```

UTF-8 is only valid for LC_CTYPE, which makes sense. This is a BSDism it looks like though, so documentation on it would be helpful. My thinking on this is that if a package really needs a non-UTF-8 encoding in 2019, they should specifically request it through setting LC_CTYPE or LC_ALL.

Reply:
Yeah, I meant by having a C.UTF-8 file generated on macOS instead of using `LC_CTYPE=UTF-8`.